src/main/java/Profile_screen/UserSearchUI.java

```diff
@@ -0,0 +1,64 @@
+package Profile_screen; /**
+ * Provides the UI elements
+ */
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.io.File;
+
+
+public class UserSearchUI implements UserPresenter {
+    private JLabel label;
+
+    public UserSearchUI() {
+        final JFrame frame = new JFrame();
+        frame.setSize(300, 100);
+        frame.setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE);
+        frame.setLayout(new FlowLayout());
+
+        final JTextField field = new JTextField("Enter a username");
+
+        field.addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                label.setText(showProfile(field.getText()));
+            }
+
+        });
+        frame.add(field);
+        label = new JLabel();
+        frame.add(label);
+        frame.setVisible(true);
+  }
+
+//  User_search_IA.UserPresenter makes UI implement showProfile to invert the use-case --> UI dependency
+    @Override
+    public String showProfile(String username) {
+        //    setting up access to the database of users:
+        UserDatabase db = new UserDatabase();
```

Comment:
I think it would be better if db is an attribute of this class, so only one UserDatabase is used, whenever this method is called